util/context_cancel_debugger.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import (
+	"fmt"
+	"runtime"
+	"time"
+
+	"github.com/juju/errors"
+	"golang.org/x/net/context"
+)
+
+var debug = false
```

Comment:
Enable be default will have some performance cost, I prefer disable by default. @hhkbp2 
Or should its value got from environment  variable?  @shenli